thorlcr/thorutil/thbuf.cpp

```diff
@@ -2171,21 +2171,21 @@ class CSharedFullSpillingWriteAhead : public CInterfaceOf<ISharedRowStreamReader
         nextInputReadEog = false;
         endOfInput = false;
     }
-    virtual void mergeStats(CRuntimeStatisticCollection & target) const override
+    virtual unsigned __int64 getStatistic(StatisticKind kind) const override
```

Comment:
I did implement it like this initially.  However,
1) the iFileIO would need to be checked to make sure it's not null.  Repeated calling to getStatistic will mean that this check is done repeatedly. 
2) inactiveStats (and later the outputStream) stats will need to be merged too.  

 Isn't it going to be less efficient to use getStatistic like this?  What is wrong with using mergeStatistic as member function? @jakesmith

Reply:
>the iFileIO would need to be checked to make sure it's not null. Repeated calling to getStatistic will mean that this check is done repeatedly.

for CSharedWriteAheadDisk must be valid after ctor, so no need to check in getStatistic

In CSharedFullSpillingWriteAhead, there'd be little to no penalty if you put a likely(iFileIO) in getStatistic and the likely is correct branch.
I don't mind so much there being a mergeStats method, but it is cleaner and follows the same pattern reusing existing code if it implements getStatistic (and makes a mergeStats impl. simple if it were needed). But it would be best to avoid the merging, getting/clearing/resetting approach regardless. With a getStatistic implementation, the mergeStats method that checks source would simply be:
```
    virtual void mergeStats(CRuntimeStatisticCollection & target) const override
    {
        if (iFileIO)
            ::mergeStats(target, this);
    }
```

but it's not likely worth it vs if (iFileIO) in getStatistic.